src/TestInfrastructure/Helpers/ProcessScope.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Diagnostics;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests.Helpers
+{
+    /// <summary>
+    /// Ensure the process is terminated when it goes out of scope.
+    /// Usage:
+    ///   using var processScope = new ProcessScope(process);
+    /// </summary>
+    /// <remarks>Disposing a process does not terminate it so we can't rely on
+    /// "using var process = ..." to clean up properly.
+    /// Instead, we'll try to fetch the process by id then "Kill" it.
+    /// </remarks>
+    public sealed class ProcessScope : IDisposable
+    {
+        private readonly int id;
+        private readonly string mainModuleFilName;
+        private readonly string processName;
+
+        public Process Process { get; }
+
+        public ProcessScope(Process process)
+        {
+            Process = process ?? throw new ArgumentNullException(nameof(process));
+
+            // Cache the properties we need now as we won't be able to access them
+            // later if the process has been disposed at the point we want to clean up.
+            id = process.Id;
+
+            (mainModuleFilName, processName) = TryGetProcessDetails(process);
+        }
+
+        public void Dispose()
+        {
+            try
+            {
+                var runningProcess = Process.GetProcessById(id);
+                Log("Process is running.  Id: " + id);
+
+                // Make sure we don't accidentally kill a newly-created process with the same id
```

Comment:
Regarding this comment: if our code is the one creating the process, wouldn't it always have the same module name and process name?

Reply:
That's the point; we don't know if it is the same process, just that it has the same id as the original process.

Checking the other properties give some additional level of certainty that we are killing the correct process, and not a newly-created process that Windows happens to have allocated the same Id.